etl-jobs/src/main/scala/org/sunbird/analytics/jobs/UserCacheIndexer.scala

```diff
@@ -0,0 +1,334 @@
+import com.redislabs.provider.redis._
+import org.apache.spark.SparkConf
+import org.apache.spark.SparkContext
+import org.sunbird.analytics.util.JSONUtils
+
+object UserCacheIndexer {
+
+  private val config: Config = ConfigFactory.load
+
+  def main(args: Array[String]): Unit = {
+
+    //val isForAllUsers = args(0) // true/false
+    var specificUserId = ""
+    var fromSpecificDate = ""
+    if (!args.isEmpty) {
+      specificUserId = args(0) // userid
+      fromSpecificDate = args(1) // date in YYYY-MM-DD format
+    }
+    val sunbirdKeyspace = config.getString("cassandra.user.keyspace")
+    val redisKeyProperty = "id" // userid
+
+    val dtf1 = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss:SSSZ")
+    val dtf2 = new SimpleDateFormat("yyyy-MM-dd")
+
+    val spark: SparkSession =
+      SparkSession
+        .builder()
+        .appName("AppName")
+        .config("spark.master", "local")
+        .config("spark.cassandra.connection.host", config.getString("spark.cassandra.connection.host"))
+        .config("spark.redis.host", config.getString("redis.host"))
+        .config("spark.redis.port", config.getString("redis.port"))
+        .config("spark.redis.db", config.getString("redis.user.database.index"))
+        .config("spark.redis.max.pipeline.size", config.getString("redis.max.pipeline.size"))
+        .getOrCreate()
+
+    def seqOfAnyToSeqString(param: Seq[AnyRef]): Seq[(String, String)]
+    = {
+      param.map {
+        case (x, y) => (x.toString, if (!y.isInstanceOf[String]) {
+          if (null != y) {
+            JSONUtils.serialize(y.asInstanceOf[AnyRef])
+          } else {
+            ""
+          }
+        } else {
+          y.asInstanceOf[String]
+        })
+      }
+    }
+
+    def filterUserData(userDF: DataFrame): DataFrame = {
+      if (null != specificUserId && StringUtils.isNotEmpty(specificUserId)) {
+        println("for specfic userId" + specificUserId)
+        userDF.filter(col("id") === specificUserId)
+      } else if (null != fromSpecificDate && StringUtils.isNotEmpty(fromSpecificDate)) {
+        println(s"Fetching all the user records from this specific date:$fromSpecificDate ")
+        userDF.filter(col("updateddate").isNull || to_date(col("updateddate"), "yyyy-MM-dd HH:mm:ss:SSSZ").geq(lit(fromSpecificDate)))
+      } else {
+        println("Input are empty hence returning userdf without filter")
+        userDF
+      }
+    }
+
+    def getUserData(): DataFrame = {
+
+      val userDF = filterUserData(spark.read.format("org.apache.spark.sql.cassandra").option("table", "user").option("keyspace", sunbirdKeyspace).load().
+        select("*").persist())
+        // Flattening the BGMS
+        .withColumn("medium", explode_outer(col("framework.medium")))
+        .withColumn("subject", explode_outer(col("framework.subject")))
+        .withColumn("board", explode_outer(col("framework.board")))
+        .withColumn("grade", explode_outer(col("framework.gradeLevel")))
```

Comment:
No need of any null check?

Reply:
Yes Santhosh, No need of null check over here.